scripts/pylib/twister/twisterlib/runner.py

```diff
@@ -296,6 +296,17 @@ def run_cmake(self, args=""):
             f'-G{self.env.generator}'
         ]
 
+        # If needed, run CMake using the package_helper script first, to only run
+        # a subset of all cmake modules. This output will be used to filter
+        # testcases, and the full CMake configuration will be run for
+        # testcases that should be built
+        if filter:
+            cmake_filter_args = [
+                f'-S{self.source_dir}',
```

Comment:
Is this necessary to add here `-S{self.source_dir}`? If I understand it correctly, the source is passed to CMake command few lines later.